lib/aggregate/attribute/string.rb

```diff
@@ -1,17 +1,53 @@
 class Aggregate::Attribute::String < Aggregate::Attribute::Builtin
 
+  class Aggregate::EncryptionError < StandardError; end;
+
   def self.available_options
     Aggregate::Attribute::Builtin.available_options + [
       :size, # The maximum length of the string
+      :encrypted, # Whether or not we need to encrypt a field
     ]
   end
 
   def load(value)
-    value.to_s
+    options[:encrypted] ? decrypt(value) : value.to_s
+  end
+
+  def decrypt(value)
+    # get string, convert to hash, read decoded iv, decode value, decrypt with iv and value
+    hash = ActiveSupport::JSON.decode value
+
+    find_decrypted_value(Base64.urlsafe_decode64(hash["encrypted_data"]), Base64.urlsafe_decode64(hash["initilization_vector"]))
+  end
+
+  def find_decrypted_value(value, iv)
+    encrypted_value = nil
+    Aggregate::Base.hashed_keys.each do |k, v|
```

Comment:
Assuming the key list is a list of keys...
```
encrypted_value = Aggregate.configuration.key_list.find do |k|
  begin
    Encryptor.decrypt(value: value, key: k, iv: iv)
  rescue OpenSSL::Cipher::CipherError
    nil
  end
end
```

Is slightly cleaner and has the added benefit of stopping early if the right key is found.